app/utils/agendaitem-utils.js

```diff
@@ -84,22 +85,42 @@ export const sortByNumber = (groupedAgendaitems, allowEmptyGroups) => {
 /**
  * Given a set of agendaitems, set their number
  * @name setAgendaitemsNumber
- * @param  {Array<agendaitem>}   agendaitems  Array of agendaitem objects to set number on.
- * @param  {Boolean} isEditor     When true, the user is allowed to edit the trigger a recalculation of the number.
- * @param {Boolean} isDesignAgenda  When true, the agenda is a designagenda.
+ * @param  {Array<Agendaitem>} agendaitems  Array of agendaitem objects to set number on.
+ * @param {Meeting} meeting The meeting the agendaitems belong to
+ * @param {Store} store The store service
+ * @param {DecisionReportGeneration} decisionReportGeneration The decisionReportGeneration service
+ * @param {Boolean} isEditor When true, the user is allowed to edit the trigger a recalculation of the number.
+ * @param {Boolean} isDesignAgenda When true, the agenda is a designagenda.
  */
-export const setAgendaitemsNumber = async(agendaitems, isEditor, isDesignAgenda) => {
+export const setAgendaitemsNumber = async(agendaitems, meeting, store, decisionReportGeneration, isEditor, isDesignAgenda) => {
   if (isEditor && isDesignAgenda) {
-    return await Promise.all(agendaitems.map(async(agendaitem, index) => {
+    const reports = [];
+    const promises = await Promise.all(agendaitems.map(async(agendaitem, index) => {
```

Comment:
We'll need to test this on large agenda's on ACCEPT. Since the operation is a bit more expensive, I wonder how this will handle an agenda of 100s of items